_posts/2015-09-24-mvvm-in-swift.markdown

```diff
@@ -0,0 +1,234 @@
+---
+layout: post
+title: "MVVM in Swift"
+date: 2015-09-24 16:13
+comments: true
+author: Ash Furrow
+github-url: https://www.github.com/ashfurrow
+twitter-url: https://twitter.com/ashfurrow
+blog-url: http://ashfurrow.com
+categories: [ios, mvvm, open source, swift, mobile]
+---
+
+Model View ViewModel has become the default way I write apps on iOS – it makes writing iOS apps a joy. I've written [about it](https://www.objc.io/issues/13-architecture/mvvm/) and [again](http://www.teehanlax.com/blog/model-view-viewmodel-for-ios/) and [again](https://speakerdeck.com/ashfurrow/reactivecocoa-at-mdevcon-2014) and [oh my](https://leanpub.com/iosfrp).
+
+But last Autumn, as our team was all-hands-on-deck to wrap up the [auctions kiosk app](http://artsy.github.io/blog/2014/11/13/eidolon-retrospective/), we chose not to use MVVM. Why not?
+
+We were building a _new app_ in a _new language_ using a _non-Swift FRP framework_. I was also teaching colleagues what they needed to know about ReactiveCocoa to help write the app. We used the MVC pattern because the relative cost of MVVM was too high.
```

Comment:
Perhaps it would be worthwhile to spell out, or put in parentheses, "Functional Reactive Programming" the first time you use it here. I think you can assume anyone reading this post is familiar with what MVC means, but FRP maybe not...